File: 700/766 - Toeplitz Matrix.c
bool isToeplitzMatrix(int** matrix, int matrixRowSize, int *matrixColSizes) {
    int matrixColSize = *matrixColSizes;

    for (int i = 0; i < matrixRowSize; ++i) {
        int val = matrix[i][0];
        for (int j = 1; j < matrixColSize && i + j < matrixRowSize; ++j) {
            if (val != matrix[i + j][j]) {
                return 0;
            }
        }
    }
    
    for (int j = 0; j < matrixColSize; ++j) {
        int val = matrix[0][j];
        for (int i = 1; i < matrixRowSize && i + j < matrixColSize; ++i) {
            if (val != matrix[i][i + j]) {
                return 0;
            }
        }
    }
    
    return 1;
}

File: csharp/766.toeplitz-matrix.cs
/*
 * @lc app=leetcode id=766 lang=csharp
 *
 * [766] Toeplitz Matrix
 */

// @lc code=start
public class Solution {
    public bool IsToeplitzMatrix(int[][] matrix) {
        int m = matrix.Length;
        if (m == 0) {
            return true;
        }
        int n = matrix[0].Length;
        for (int i = 0; i < m; ++i) {
            int val = matrix[i][0];
            for (int r = i + 1, c = 1; r < m && c < n; ++r, ++c) {
                if (val != matrix[r][c]) {
                    return false;
                }
            }
        }
        for (int j = 1; j < n; ++j) {
            int val = matrix[0][j];
            for (int r = 1, c = j + 1; r < m && c < n; ++r, ++c) {
                if (val != matrix[r][c]) {
                    return false;
                }
            }
        }
        return true;
    }
}
// @lc code=end

